File: 2021/src/day01/day01.py
input = [int(x) for x in open("input.txt", "r").read().splitlines()]


def part1():
    lastdepth = 0
    increases = 0
    for depth in input[1:]:
        if depth > lastdepth:
            increases += 1
        lastdepth = depth
    print("Part 1")
    print("  ", increases)


def part2():
    lastdepth = 0
    increases = 0
    for i in range(1, len(input) - 3):
        depth = sum(input[i : i + 3])
        if depth > lastdepth:
            increases += 1
        lastdepth = depth
    print("Part 2")
    print("  ", increases)


part1()
part2()

File: 2017/src/day01/day01.js
const input = (raw) => raw.split("").map(Number);

export const part1 = (raw) => {
  const data = input(raw);
  let sum = 0;
  for (let i = 0; i < data.length; i += 1) {
    if (data[i] === data[(i + 1) % data.length]) {
      sum += data[i];
    }
  }
  return sum;
};

export const part2 = (raw) => {
  const data = input(raw);
  const skip = data.length / 2;

  let sum = 0;
  for (let i = 0; i < data.length; i += 1) {
    if (data[i] === data[(i + skip) % data.length]) {
      sum += data[i];
    }
  }
  return sum;
};
